src/main/java/frc/robot/subsystems/VisionSubsystem.java

```diff
@@ -123,18 +118,39 @@ public VisionSubsystem(DrivebaseWrapper aprilTagsHelper) {
   }
 
   public void update() {
-    latestResult = photonCamera.getLatestResult();
-    latestPose = photonPoseEstimator.update(latestResult);
-    lastRawTimestampSeconds = latestResult.getTimestampSeconds();
-
-    if (latestPose.isPresent()) {
-      lastTimestampSeconds = latestPose.get().timestampSeconds;
-      lastFieldPose = latestPose.get().estimatedPose.toPose2d();
-      rawVisionFieldObject.setPose(lastFieldPose);
-      // gonna fix this later
 
-      aprilTagsHelper.addVisionMeasurement(lastFieldPose, lastTimestampSeconds, STANDARD_DEVS);
-      d2f.setRobotPose(aprilTagsHelper.getEstimatedPosition());
+    for (PhotonPipelineResult result : photonCamera.getAllUnreadResults()) {
+      latestResult = result;
+      latestPose = photonPoseEstimator.update(latestResult);
+      lastRawTimestampSeconds = latestResult.getTimestampSeconds();
+      System.out.println(latestResult);
+
+      if (latestPose.isPresent()) {
+        if (lastRawTimestampSeconds > lastTimestampSeconds) {
```

Comment:
Combine these 2 if's into one statement:

if (latestPost.isPresent() && lastRawTimestampSeconds > lastTimeTimestampSeconds) {

Reply:
Alternatively move the first if() up and skip getting all of the other variables until after you know isPresent() is true.  This will save you some execution time.

for (PhotonPipelineResult result : photonCamera.getAllUnreadResults()) {
      latestResult = result;
      latestPose = photonPoseEstimator.update(latestResult);
      if (latestPose.isPresent()) {
        lastRawTimestampSeconds = latestResult.getTimestampSeconds();
        System.out.println(latestResult);
        if (lastRawTimestampSeconds > lastTimestampSeconds) {